From fa57f929efb6307103e7d74bcb3c3737decb8a2c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A9ven=20Car?= <meven@kde.org>
Date: Wed, 17 Jul 2024 12:41:14 +0200
Subject: [PATCH] kastatsfavoritesmodel: Fix a crash

Regression of b5ce9cb4

(cherry picked from commit e01b885a1f2d403e9ecea74d2a059311e37876dd)
---
 applets/kicker/plugin/kastatsfavoritesmodel.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/applets/kicker/plugin/kastatsfavoritesmodel.cpp b/applets/kicker/plugin/kastatsfavoritesmodel.cpp
index 75d37a0a46f..6638971a73c 100644
--- a/applets/kicker/plugin/kastatsfavoritesmodel.cpp
+++ b/applets/kicker/plugin/kastatsfavoritesmodel.cpp
@@ -193,9 +193,11 @@ public:
                 // https://crash-reports.kde.org/organizations/kde/issues/23450/
                 const auto itemEntries = m_itemEntries;
                 for (auto it = itemEntries.cbegin(); it != itemEntries.cend(); it = std::next(it)) {
-                    it.value()->reload();
-                    if (it.value() && !it.value()->isValid()) {
-                        keys << it.key();
+                    if (it.value()) {
+                        it.value()->reload();
+                        if (!it.value()->isValid()) {
+                            keys << it.key();
+                        }
                     }
                 }
                 if (!keys.isEmpty()) {
-- 
GitLab

